test/system/uniswap-v2.test.js

```diff
@@ -0,0 +1,144 @@
+    await assetPool.connect(governance).transferOwnership(coveragePool.address)
+
+    uniswapV2Router = await ethers.getContractAt(
+      "IUniswapV2Router",
+      uniswapV2RouterAddress
+    )
+
+    const SignerBondsUniswapV2 = await ethers.getContractFactory(
+      "SignerBondsUniswapV2"
+    )
+    signerBondsUniswapV2 = await SignerBondsUniswapV2.deploy(
+      uniswapV2Router.address,
+      coveragePool.address
+    )
+    await signerBondsUniswapV2.deployed()
+
+    // Simulate that risk manager deposits signer bonds on the Uniswap strategy.
+    await signerBondsUniswapV2
+      .connect(riskManager)
+      .swapSignerBonds({ value: ethers.utils.parseEther("20") })
+  })
+
+  describe("test initial state", () => {
+    describe("asset pool", () => {
+      it("should not have any collateral tokens", async () => {
+        expect(await collateralToken.balanceOf(assetPool.address)).to.equal(0)
+      })
+    })
+
+    describe("swap strategy", () => {
+      it("should have the signer bonds deposited", async () => {
+        const balance = await (
+          await ethers.getSigner(signerBondsUniswapV2.address)
+        ).getBalance()
+
+        expect(balance).to.equal(ethers.utils.parseEther("20"))
+      })
+    })
+  })
+
+  describe("when signer bonds are swapped on UniswapV2", () => {
+    let tx
+
+    before(async () => {
+      tx = await signerBondsUniswapV2
+        .connect(thirdParty)
+        .swapSignerBondsOnUniswapV2(ethers.utils.parseEther("10"))
+    })
+
+    it(
+      "should take the swapped amount from the swap strategy " +
+        "contract balance",
+      async () => {
+        await expect(tx).to.changeEtherBalance(
+          signerBondsUniswapV2,
+          ethers.utils.parseEther("-10")
+        )
+      }
+    )
+
+    it("should send acquired tokens to the asset pool", async () => {
+      // WETH_in = 10000000000000000000 (10 ETH)
+      // WETH_in_with_fee = 9970000000000000000 (WETH_in * 99.7%)
+      // WETH_reserve = 1196274582417900623439 (1196 WETH)
+      // KEEP_reserve = 10799279239931490639501248 (10799279 KEEP)
+      //
+      // To calculate the KEEP output amount we must do:
+      // WETH_in_with_fee * KEEP_reserve / (WETH_reserve + WETH_in_with_fee)
+      // which equals to 89259521320540407673397 (89259 KEEP). The exact amount
+      // is obtained because there is no slippage in test environment.
+      expect(await collateralToken.balanceOf(assetPool.address)).to.equal(
+        "89259521320540407673397"
+      )
+    })
+
+    it("should emit UniswapV2SwapExecuted event", async () => {
+      await expect(tx)
+        .to.emit(signerBondsUniswapV2, "UniswapV2SwapExecuted")
+        .withArgs([
+          "10000000000000000000", // ETH -> WETH
+          "89259521320540407673397", // WETH -> KEEP
+        ])
+    })
+
+    it("should consume a reasonable amount of gas", async () => {
+      await expect(parseInt(tx.gasLimit)).to.be.lessThan(11472000)
```

Comment:
:scream: Something is wrong, block gas limit is around 12M now. 

Assuming 50Gwei gas price, one would need to have ~0.57 ETH on their account to execute this transaction.

Reply:
There is something wrong with the estimate provided by Hardhat. Please see https://github.com/keep-network/coverage-pools/pull/66/commits/c3fda9c131035ca3d446b0575d526a942b6b1d38.